ITE/5/README.md

```diff
@@ -0,0 +1,343 @@
+# ITE-5: Signature scheme that avoids canonicalization
+
+<table>
+<tr><td>ITE<td>5
+<tr><td>Title<td>Signature scheme that avoids canonicalization
+<tr><td>Sponsor<td><a href="https://github.com/santiagotorres">Santiago Torres-Arias</a>
+<tr><td>Status<td>Draft 💬
+<tr><td>Type<td>Standards
+<tr><td>Created<td>2020-09-28
+</table>
+
+# Abstract
+
+This [in-toto enhancement (ITE)](https://github.com/in-toto/ITE) proposes
+changing the in-toto/TUF signature scheme to (a) avoid relying on
+canonicalization for security and (b) reduce the possibility of
+misinterpretation of the payload. The serialized payload is encoded as a string
+and verified by the recipient before deserializing. A backwards compatible
+variant is available.
+
+# Specification
+
+## Current format
+
+The
+[current signature format](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#42-file-formats-general-principles)
+has a BODY that is a regular JSON object and a signature over the
+[Canonical JSON] serialization of BODY.
+
+```json
+{
+  "signed": <BODY>,
+  "signatures": [{
+    …,
+    "sig": "<Hex(Sign(CanonicalJson(BODY)))>"
+  }, …]
+}
+```
+
+To verify, the consumer parses the whole JSON file, re-serializes BODY using
+Canonical JSON, then verifies the signature.
+
+## Proposed format
+
+Instead of relying on Canonical JSON, the producer records the signed bytes
+exactly as signed and the consumer verifies those exact bytes before parsing. In
+addition, the signature now includes an authenticated `payloadType` field
+indicating how to interpret the payload.
+
+```json
+{
+  "payload": "<Base64(SERIALIZED_BODY)>",
+  "payloadType": "<PAYLOAD_TYPE>",
+  "signatures": [{
+    …,
+    "sig": "<Base64(Sign(PAE([PAYLOAD_TYPE, SERIALIZED_BODY])))>"
+  }, …]
+}
+```
+
+where PAE is the
+[PASETO Pre-Authentication Encoding](https://github.com/paragonie/paseto/blob/master/docs/01-Protocol-Versions/Common.md#authentication-padding):
+
+```none
+PAE([type, body]) := le64(2) || le64(len(type)) || type || le64(len(body)) || body
+le64(n) := 64-bit little-endian encoding of `n`, where 0 <= n < 2^63
+```
+
+The PAYLOAD_TYPE is a URI indicating how to interpret SERIALIZED_BODY. It
+encompasses the content type (JSON, CBOR, etc.), the purpose, and the schema
+version of the payload. This obviates the need for the `_type` field within
+in-toto/TUF payloads. Examples:
+
+-   https://in-toto.to/Link/v0.9
+-   https://in-toto.to/Layout/v0.9
+-   https://theupdateframework.com/Root/v1.0.5
+-   etc...
+
+The switch from Hex to Base64 for `sig` is to save space and to be consistent
+with `payload`.
+
+## Steps
+
+To sign:
+
+-   Serialize BODY according to PAYLOAD_TYPE. Call the result SERIALIZED_BODY.
+-   Sign PAE([PAYLOAD_TYPE, PAYLOAD]), base64-encode the result, and store it in
```

Comment:
PAYLOAD is not defined above. Did you mean: SERIALIZED_BODY? Ditto under "to verify" below.

I think that using PAYLOAD here would be confusing, because there already is a JSON field called "payload" and that field contains a transformation of SERIALIZED_BODY.

Reply:
Oops, thanks. Fixed.